lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,248 @@
+
+  /**
+   * Here, we tell Renovate that this data source can respect multiple upstream repositories
+   */
+  override readonly registryStrategy = 'merge';
+
+  requiredPackageKeys = ['Package', 'Version', 'Homepage'];
+
+  initCacheDir(cfg: DebLanguageConfig): void {
+    mkdirSync(cfg.deb.downloadDirectory, { recursive: true });
+    mkdirSync(cfg.deb.extractionDirectory, { recursive: true });
+  }
+
+  async runCommand(wd: string, command: string): Promise<void> {
+    const pexec = promisify(exec);
+    logger.trace('running command ' + command + ' in directory ' + wd);
+    const { stdout, stderr } = await pexec(
+      ['cd', '"' + wd + '"', '&&', command].join(' ')
+    );
+    logger.debug(stdout);
+    logger.debug(stderr);
+  }
+
+  async downloadAndExtractPackage(
+    cfg: DebLanguageConfig,
+    packageUrl: string
+  ): Promise<string> {
+    // we hash the package URL and export the hex to make it file system friendly
+    // we use the hashed url as the filename for the local directories/files
+    const hash = createHash('sha256');
+    hash.update(packageUrl);
+    const hashedPackageUrl = hash.copy().digest('hex');
+    const downloadLocation = cfg.deb.downloadDirectory + '/' + hashedPackageUrl;
+    const compressedFile = downloadLocation + '/Packages.xz';
+    const extractedFile =
+      cfg.deb.extractionDirectory + '/' + hashedPackageUrl + '.txt';
+
+    // curl will not modify the local file, if the upstream HTTP server
+    // does not return a modified version. We can use this information
+    // to not re-extract the same file again. For this to work we need
+    // the current modification timestamp of the local package file.
+    let lastTimestamp = -1;
+    try {
+      const stats = statSync(compressedFile);
+      lastTimestamp = stats.mtime.getTime();
+    } catch (e) {
+      // ignore if the file doesnt exist
+    }
+
+    logger.debug(
+      'Downloading package file from ' + packageUrl + ' as ' + downloadLocation
+    );
+    /**
+     * curl -o "$file" -z "$file" "$uri"
+     */
+    mkdirSync(downloadLocation, { recursive: true });
+    await this.runCommand(
+      downloadLocation,
+      ['curl', '-o', 'Packages.xz', '-z', 'Packages.xz', packageUrl].join(' ')
+    );
+
+    const stats = statSync(compressedFile);
+    const newTimestamp = stats.mtime.getTime();
+
+    if (newTimestamp <= lastTimestamp && existsSync(extractedFile)) {
+      logger.debug(
+        "No need to extract file as wget didn't update the file and it exists"
+      );
+      return extractedFile;
+    }
+
+    logger.debug(
+      'Extracting package file ' + downloadLocation + ' to ' + extractedFile
+    );
+    /**
+     * --threads=0 use all available cores
+     * -k keep the downloaded file (needed by wget to not redownload)
+     * -c print to stdout (we pipe)
+     * -d decompress
+     */
+    await this.runCommand(
+      downloadLocation,
+      ['xz --threads=0 -k -c -d', 'Packages.xz', '>', extractedFile].join(' ')
```

Comment:
We should use a javascipt based extraction, see terraform manager

Reply:
Done as well